Engine/Client/Src/GraphicalLoad/GraphicalLoad.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+** EPITECH PROJECT, 2024
+** R-Type Client
+** File description:
+** GraphicalLoad
+*/
+
+#include <memory>
+#include <string>
+#include <filesystem>
+
+#include "IGraphic.hpp"
+#include "DLLoader.hpp"
+#include "GraphicalLoad.hpp"
+
+#define PATH_GRAPHIC_LIB "Engine/Client/Lib/Graphic/"
+
+static std::string getPathGraphicalLib()
+{
+    std::string path = PATH_GRAPHIC_LIB;
+
+    for (const auto& file : std::filesystem::directory_iterator(path)) {
+        std::string prefix = path + "libgraphic_";
+        std::string suffix = ".so";
+        if (file.path().string().length() < prefix.size() + suffix.size())
+            continue;;
+        if (std::string(file.path().string()).compare(std::string(file.path().string()).length() - suffix.length(), suffix.length(), suffix) == 0 && std::string(file.path().string()).compare(0, prefix.length(), prefix) == 0)
```

Comment:
On GitHub review, It's unreadable, I think you can consider using .find() because It's clear and simpler in my opinion